modules/api_gateway/main.tf

```diff
@@ -105,26 +97,34 @@ resource "aws_iam_role_policy" "invocation_policy" {
       {
         "Action" = "lambda:InvokeFunction",
         "Effect" = "Allow",
-        "Resource" = distinct([
-          # use wildcard for versioned lambdas to avoid
-          # revoking permissions on previous version during deploy
-          for lambda_key, value in var.lambdas : replace(value.function_arn, "/:\\d+$/", ":*")
-        ])
+        "Resource" = [for e in distinct(
+          concat([
+            # use wildcard for versioned lambdas to avoid
+            # revoking permissions on previous version during deploy
+            for key, value in var.authorizers : replace(value.function_arn, "/:\\d+$/", ":*") if lookup(value, "function_arn", "") != ""
+            ], [
+            for key, value in var.lambdas : replace(value.function_arn, "/:\\d+$/", ":*")
+          ])
+        ): e]
       }
     ]
   })
 }
 
 resource "aws_api_gateway_authorizer" "authorizer" {
-  for_each = local.authorizer_keys
+  for_each = var.authorizers
 
-  name        = "api-authorizer-${each.key}"
-  rest_api_id = aws_api_gateway_rest_api.rest_api.id
-  type        = lookup(var.lambdas[each.key], "authorizer_type", "TOKEN")
-  authorizer_uri                   = var.lambdas[each.key]["function_invoke_arn"]
-  authorizer_credentials           = aws_iam_role.invocation_role.arn
-  identity_source                  = lookup(var.lambdas[each.key], "identity_source", "method.request.header.X-Auth-Token")
-  authorizer_result_ttl_in_seconds = parseint(lookup(var.lambdas[each.key], "authorizer_result_ttl_in_seconds", "900"), 10)
+  name                             = "api-authorizer-${each.key}"
+  rest_api_id                      = aws_api_gateway_rest_api.rest_api.id
+  type                             = lookup(each.value, "authorizer_type", "TOKEN")
+  authorizer_uri                   = each.value["function_invoke_arn"]
+  authorizer_credentials           = (lookup(each.value, "function_arn", "") != "" ? aws_iam_role.invocation_role.arn : null)
```

Comment:
Might be nice if the check against `""` was explicit, e.g. there could be a `map` in the `locals` which has a mapping of the authorizer `key` and bool of whether or not it's a cross-account invocation. That would make the logic a lot clearer and harder to miss.

Reply:
Because of how Terraform typing works, it's easier to work with maps of strings instead of having one attribute value be a different type (boolean). I did abstract out the logic to a local to clean up the authorizer resource a bit though.